### PR TITLE
[ci][test] Modifications to static analysis contribution via Tomasbjerre

### LINKED ISSUE #33: Discuss static analysis options
##### Description

I raised the question in #26 about how we could verify standard usage of slf4j loggers at compile time, and whether there is a static analysis tool we could use.  We could look into SonarQube, but I've also found a Findbugs plugin at https://github.com/KengoTODA/findbugs-slf4j which would do exactly this.

Should we consider incorporating this type of static analysis now? With so many open source contributors, it would be helpful to standardize on some coding patterns to reduce maintenance overhead.

##### openapi-generator version

all

##### OpenAPI declaration file content or url

n/a

##### Command line used for generation

n/a

##### Steps to reproduce

n/a

##### Related issues/PRs

* #26 

##### Suggest a fix/enhancement

Incorporate FindBugs or similar into the build.

### PR DESCRIPTION
**Pull request by jimschubert**
This extends on submission by @tomasbjerre in #5040 by moving the PMD and Spotbugs  plugin execution to a separate profile `static-analysis`. This keeps CheckStyle in the default build because it added a very minimal amount of time to execution.

Once merged, I'd like to extend the Sonar CI GitHub Action to be a static analysis action which caches the xml reports generated by these plugins as build artifacts.  This will give us SonarCloud as a means for evaluating the static analysis, and allow for contributors to download the `*.xml` reports from the action. Once the GitHub action is updated, we can also close #33 as well.

Core contributors may run static analysis with:

```
mvn -Pstatic-analysis install
```

The analysis is separated from default functionality to reduce impact to community contributions. SpotBugs/PMD may add a non-trivial amount of time to builds on some machines.

cc @OpenAPITools/generator-core-team 

Closes #5040

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: spotbugs-exclude.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+			<And>
+				<Class name="io.undertow.server.protocol.ajp.AjpRequestParseState" />
+				<Field name="dataSize" />
+			</And>
+		</Or>
+	</Match>
+
+	<!-- For source/class naming, see https://spotbugs.readthedocs.io/en/stable/filter.html#java-element-name-matching -->
+	<Match>
+		<!-- Ignore all source files with samples in path name -->
```
Comment (tomasbjerre) [inline spotbugs-exclude.xml]: One could argue that the generated code should also be free from issues found by static code analysis. I'm not sure about excluding samples here.

I can see how users have different code standards and running Checkstyle and PMD is not worth the effort, but I'm not sure it is good to exclude Spotbugs here.

File: pom.xml [hunk 2]
```diff
@@ -470,6 +542,73 @@
...
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>com.github.spotbugs</groupId>
+                        <artifactId>spotbugs-maven-plugin</artifactId>
+                        <version>${spotbugs.plugin.version}</version>
+                        <configuration>
+                            <failOnError>false</failOnError>
+                            <!-- https://spotbugs.readthedocs.io/en/stable/effort.html -->
+                            <effort>min</effort>
```
Comment (tomasbjerre) [inline pom.xml]: Now that it does not run by default, perhaps it does not need to be `min`?

**Comment by tomasbjerre** [comment]
The `travis.yml` can be extended to include this tool:
https://www.npmjs.com/package/violation-comments-to-github-command-line

But perhaps you would rather have Sonar do the commenting.

File: spotbugs-exclude.xml (hunk 1 above)
Comment (jimschubert) [reply spotbugs-exclude.xml]: Another reason to exclude these here is that I assume most users use custom templates. Analyzing our "samples" is therefore very much like analyzing our tests.

Comment (jimschubert) [reply spotbugs-exclude.xml]: Another reason to exclude these here is that I assume most users use custom templates. Analyzing our "samples" is therefore very much like analyzing our tests.

File: pom.xml (hunk 2 above)
Comment (jimschubert) [reply pom.xml]: I can evaluate this once it's hooked into the target pipeline.

File: spotbugs-exclude.xml (hunk 1 above)
Comment (jimschubert) [reply spotbugs-exclude.xml]: Another concern is that only a very small subset of our samples are Java :\
Maybe we could later work out the pattern to exclude everything but java client and server outputs in the samples directory.

**Comment by jimschubert** [comment]
> The `travis.yml` can be extended to include this tool:
> [npmjs.com/package/violation-comments-to-github-command-line](https://www.npmjs.com/package/violation-comments-to-github-command-line)
> 
> But perhaps you would rather have Sonar do the commenting.

We have some cleanup to do with respect to how much is being logged in Travis. As it is, both Travis and CirciCI UIs lock up my browser because of the amount of text output. I'm looking at modifying our logger to reduce duplicated log lines, and once that's done I will look out the outputs in other CI.

I'm thinking the `static-analysis` profile may also be run in CircleCI for contributor outputs.

**Pull request merged**